lib/hammer_cli_foreman/sessions.rb

```diff
@@ -0,0 +1,101 @@
+require 'uri'
+module HammerCLIForeman
+  class Sessions
+
+    STORAGE_DIR = '~/.hammer/sessions/'
+
+    def self.get(url)
+      raise _('Sessions are not enabled, please check your Hammer settings.') unless enabled?
+      unless File.exist?(storage)
+        FileUtils.mkdir_p(storage, mode: 0700)
+      end
+      session_filename = session_file(url)
+      if !ensure_mode(storage, '40700') || !ensure_mode(session_filename, '100600')
+        warn _("Using session auth with invalid permissions on session files is not recommended.")
+      end
+      HammerCLIForeman::Session.new(session_filename)
+
+    #   if File.exist?(session_storage)
+    #     session_data = JSON.parse(File.read(session_storage))
+    #     return nil if session_data['user_name'].nil? && session_data['session_id'].nil?
+    #     @user = session_data['user_name']
+    #     @session_id = session_data['session_id']
+    #     @auth_type = session_data['auth_type']
+    #   end
+    # rescue JSON::ParserError
+    #   warn _('Invalid session file format.')
+    #   destroy_session
+    #   nil
+
+    end
+
+    def self.storage
+      STORAGE_DIR
+    end
+
+    def self.enabled?
+      HammerCLI::Settings.get(:foreman, :use_sessions)
+    end
+
+    def self.session_file(url)
+      raise _('The url is empty. Session can not be created.') if url.nil? || url.empty?
+      uri = URI.parse(url)
+      File.join(storage, "#{uri.scheme}_#{uri.host}")
+    rescue URI::InvalidURIError
+      raise _('The url (%s) is not a valid URL. Session can not be created.') % url
+    end
+
+    private
+
+    def self.ensure_mode(file, expected_mode)
+      return true unless File.exist?(file)
+      mode = File.stat(file).mode.to_s(8)
+      if mode != expected_mode
+        warn _("Invalid permissions for %{file}: %{mode}, expected %{expected_mode}.") % {
+          :mode => mode,
+          :expected_mode => expected_mode,
+          :file => file
+        }
+        false
+      else
+        true
+      end
+    end
+  end
+
+  class Session
+    attr_accessor :id, :user_name, :auth_type
+
+    def initialize(session_path)
+      @session_path = session_path
```

Comment:
The session_path should be expanded otherwise the session is stored in CWD
```ruby
session_path = File.expand_path(session_path)
```

Reply:
It is better to get corrected above so that STORAGE_DIR is expanded since beginning